tests/rst_output_pytest.py

```diff
@@ -267,7 +267,7 @@
     ":gherkin-scenario-keyword:`Scenario:` :gherkin-scenario-content:`Test a Scenario`\n",
     "---------------------------------------------------------------------------------\n\n",
     ".. pull-quote::\n\n",
-    "    :gherkin-tag-keyword:`Tagged:` quarantined, JIRA-1234\n\n",
+    "    :gherkin-tag-keyword:`Tagged:` JIRA-1234, quarantined\n\n",
```

Comment:
This was an unexpected change. I can't currently explain why the order changed, but the tests keep passing over and over, so I'm sure it isn't random (if it is, I should play the lottery getting a coin to land on heads that many times in a row!)

Reply:
Since we expected to see the justification/Ticket tags after the non-operational status tags, this is concerning.
I agree, I don't see anything in this PR that would account for the change though.

Reply:
I checked and the behave output files always have the quarantined tag first.
But this isn't the only place where the pytest tags are out of order.
While this bugs me, so long as it stays pytest only, I am willing to defer the urgency of fixing it, in this PR, to those directly affected.

Reply:
Yes that's a good point. I'll try to take a closer look to figure this out. Maybe @bradsbrown has a hunch.

Reply:
"Having slept on it", my "guess" is that the pytest bdd parser does not consider the order of the tags to be important and is not using an order preserving data type to accumulate them. Given that, IIRC, it doesn't even support tags on multiple lines, I would not be surprised to see something like-ish `set(tag.strip() for tag in tag_line.split())`
Why that would suddenly change though is more curious than why it was never ordered in the first place. If the underlying mechanism is not ordered, then doing anything about it is out of scope for this PR.

tl;dr this only affects pytest so I am willing to sign off because behave is still working properly.

Reply:
That's close enough to verbatim [how pytest-bdd gets tags](https://github.com/pytest-dev/pytest-bdd/blob/master/pytest_bdd/feature.py#L142).

Reply:
For the record, I'm guessing the expected result file changed because you had other updates to make in it, but the reason the tests consistently pass regardless of tag order is that we [intentionally compare tags as a set](https://github.com/jolly-good-toolbelt/sphinx_gherkindoc/blob/master/tests/test_pytest_writer.py#L46) between the actual and expected so that we can be as apathetic about tag order as `pytest-bdd` has chosen to be.

Reply:
That explains it! So this change really has no effect one way or another. It might be worth noting somewhere that tag order is not guaranteed when using pytest-bdd parser, but right now no one we know is relying on that. If there's an obvious place to put it, I'll drop it in here, but otherwise I'll FF.